rfcs/DL-2023-02-27-01-portal-host-detection.md

```diff
@@ -0,0 +1,60 @@
+
+# RFC - Portal Host Detection
+
+Feb 25, 2023
+
+
+## Question
+
+What is the best user experience we can create within the next month?
+
+
+## Context
+
+When an Agent is completely offline, it will take the full Holochain timeout to get a response.
+Holochain currently supports a vector of calls so that a list of potential hosts can be queried all
+at once.  This still results in the full wait for the timeout if any of the hosts are offline.
+
+This is because QUIC does not differentiate between the connection timeout and the request timeout.
+So this problem should go away with WebRTC.
+
+
+## Options
```

Comment:
Another option would be similar to Ping/Pong vector call, but driving all the calls from the UI side. App store would implement a zome function that given an agent public key of a host, it's tries to call_remote to that host, and another zome function that returns the list of hosts available to the UI. Then, the UI would initiate one async call per host to try to connect to each of them, and cancel all of them as soon as some answer gets returned.